Mlem/Models/Trackers/Feed/Generic Multi-Trackers/ChildTracker.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  ChildTracker.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-10-16.
+//
+
+import Foundation
+
+class ChildTracker<Item: ChildTrackerItem>: BasicTracker<Item>, ChildTrackerProtocol {
+    private var parentTracker: ParentTracker<Item.ParentType>?
```

Comment:
Does a reference cycle happen here? The swift compiler doesn't check for reference cycles iirc, so this might cause a memory leak? Perhaps a weak reference is needed?

```suggestion
    private weak var parentTracker: ParentTracker<Item.ParentType>?
```

Reply:
Nice catch, that's absolutely a reference cycle